micro_dl/cli/preprocess_script.py

```diff
@@ -31,185 +35,383 @@ def parse_args():
+    if flat_field_dir is not None:
+        correct_flat_field = True
+    mask_processor_inst.generate_masks(
+        correct_flat_field=correct_flat_field,
+        str_elem_radius=str_elem_radius,
+    )
+    mask_dir = mask_processor_inst.get_mask_dir()
+    mask_out_channel = mask_processor_inst.get_mask_channel()
+    return mask_dir, mask_out_channel
+
+
+def tile_images(params_dict,
+                tile_dict,
+                resize_flag,
+                flat_field_dir,
+                mask_dir,
+                mask_out_channel):
+    """Tile images
+
+    :param dict params_dict: dict with keys: input_dir, output_dir, time_ids,
+     channel_ids, pos_ids, slice_ids, int2strlen, uniform_struct, num_workers
+    :param dict tile_dict: dict with tiling related keys: tile_size, step_size,
+     image_format, depths. optional: min_fraction, mask_channel, mask_dir,
+     mask_depth, tile_3d
+    :param bool resize_flag: indicator if resize related params in pp_config
+    :param str flat_field_dir: dir with flat field correction images
+    :param str mask_dir: dir with masks (from generate_masks)
+    :param int mask_out_channel: channel assigned to generated masks
+    :return str tile_dir: dir with tiled images
+    """
+
+    kwargs = {'input_dir': params_dict['input_dir'],
+              'output_dir': params_dict['output_dir'],
+              'tile_dict': tile_dict,
+              'time_ids': params_dict['time_ids'],
+              'slice_ids': params_dict['slice_ids'],
+              'channel_ids': params_dict['channel_ids'],
+              'pos_ids': params_dict['pos_ids'],
+              'flat_field_dir': flat_field_dir,
+              'num_workers': params_dict['num_workers'],
+              'int2str_len': params_dict['int2strlen']}
+
+    if params_dict['uniform_struct']:
+        if 'tile_3d' in tile_dict:
+            if resize_flag:
+                warnings.warn(
+                    'If resize_3d was used, slice_idx corresponds to start'
+                    'slice of each volume.If slice_ids=-1, the slice_ids'
+                    'will be read from frames_meta.csv. Assuming slice_ids'
+                    'provided here is fixed for these gaps.', Warning)
+            tile_inst = ImageTilerUniform3D(**kwargs)
+        else:
+            tile_dict['tile_3d'] = False
+            tile_inst = ImageTilerUniform(**kwargs)
+    else:
+        # currently not supported but should be easy to extend
+        tile_dict['tile_3d'] = False
+        tile_inst = ImageTilerNonUniform(**kwargs)
+
+    tile_dir = tile_inst.get_tile_dir()
+
+    # retain tiles with a minimum amount of foreground
+    if 'min_fraction' in tile_dict:
+        min_fraction = tile_dict['min_fraction']
+        mask_depth = 1
+
+        if mask_out_channel is None:
+            assert 'mask_channel' in tile_dict, \
+                'Mask channel is not provided'
+            mask_out_channel = tile_dict['mask_channel']
+
+        if 'mask_dir' is None:
```

Comment:
Could you please elaborate? mask_dir gets set to None in https://github.com/czbiohub/microDL/blob/7dccbb3dda897b9c0855f3b75663b96c3e3d5e94/micro_dl/cli/preprocess_script.py#L300 and if masks is not in config it will remain None. Wondering where I got this wrong?